From 0a9c9fba2fbf38508f551266b7ea51ed562dbb13 Mon Sep 17 00:00:00 2001
From: Gregory Norton <gregory.norton@me.com>
Date: Fri, 18 Sep 2026 20:06:47 -0400
Subject: [PATCH 05/11] MouseSettings: Point at the host gpu.h shim

MouseWidget includes <WindowServer/Screen.h> (for the mouse acceleration
limits), which reaches <sys/devices/gpu.h>; on host builds that header is
provided by the shim generated by the WindowServer target.
---
 Userland/Applications/MouseSettings/CMakeLists.txt | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/Userland/Applications/MouseSettings/CMakeLists.txt b/Userland/Applications/MouseSettings/CMakeLists.txt
index d71686ead2..7f98c3c1f8 100644
--- a/Userland/Applications/MouseSettings/CMakeLists.txt
+++ b/Userland/Applications/MouseSettings/CMakeLists.txt
@@ -21,4 +21,10 @@ set(SOURCES
 )
 
 serenity_app(MouseSettings ICON app-mouse)
+if (NOT SERENITYOS)
+    # MouseWidget includes <WindowServer/Screen.h> (for the mouse acceleration
+    # limits), which reaches <sys/devices/gpu.h>; on host builds that header is
+    # provided by the shim generated by the WindowServer target.
+    target_include_directories(MouseSettings PRIVATE ${CMAKE_BINARY_DIR}/lagom/Userland/Services/WindowServer/include)
+endif()
 target_link_libraries(MouseSettings PRIVATE LibCore LibGfx LibGUI LibIPC LibMain)
-- 
2.55.0

